src/awesome_map/BoundaryInterceptor.js

```diff
@@ -151,6 +166,39 @@ define(function(require) {
         this._easing = options.easing || EasingFunctions.easeOutQuart;
 
         /**
+         * Defines the minimum number of pixels beyond the boundary which a scroll event
+         * must reach for boundary events to be fired.
+         * @type {number}
+         * @private
+         */
+        this._boundarySensitivity = options.boundarySensitivity || DEFAULT_BOUNDARY_SENSITIVITY;
+
+        /**
+         * Observable for subscribing to scroll events beyond AwesomeMap boundaries.
+         *
+         * @method AwesomeMap#onScrollPastTopBoundary
+         * @param {Function} callback
+         *        Invoked with (sender, {
+         *            boundary: {@link BoundaryTypes}
+         *        })
+         */
+        this.onScrollPastBoundary = Observable.newObservable();
+
+        /**
+         * A list of boundaries that have been in view since the last completed user event 
+         * @type {Array.<BoundaryType>}
+         * @private
+         */
+        this._visibleBoundaries = [];
+        // Initialize with all of the boundaries, and any that are not in view will be removed
+        // when an event is processed.
+        for (var boundary in BoundaryTypes) {
+            if (BoundaryTypes.hasOwnProperty(boundary)) {
```

Comment:
This is not essential, since BoundaryTypes does not inherit from anything, but we might as well keep it.